tripal/src/TripalField/TripalFormatterBase.php

```diff
@@ -25,4 +32,79 @@ abstract class TripalFormatterBase extends FormatterBase {
   public function sanitizeKey($key) {
     return preg_replace('/[^\w]/', '_', $key);
   }
+
+  /**
+   * {@inheritdoc}
+   *
+   * Initializes value of max_delta
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $max_delta = \Drupal::config('tripal.settings')->get('tripal_entity_type.publish_global_max_delta');
+    // max_delta defaults to 100 if not defined
+    if (is_null($max_delta) or (trim($max_delta) === '')) {
+      $max_delta = 100;
+    }
+    $this->max_delta = $max_delta;
+  }
+
+  /**
+   * Converts a list of multiple renderable items into a list.
+   * A single item is passed through unchanged.
+   *
+   * @param array $list
+   *   A list of renderable items
+   * @return array
+   *   A render array. Will be empty if there were no items.
```

Comment:
Maybe expand this a bit to include mention of the warning?

```suggestion
   * HELPER: Makes multiple renderable items into a list.
   *
   * Notes: If the number of items exceeds the max delta for this field
   * then a warning will be added right before the list.
   *
   * @param array $list
   *   A list of renderable items
   * @return array
   *   A render array where (1) if no items then an empty element is
   *   returned, (2) if there is one item it is passed through unchanged,
   *   (3) if `$items < $max_delta` then a list of items is returned, (4) 
   *   if `$items > $max_delta` then two elements are returned where the
   *   first is a warning string and the second is a list of items.
```

Reply:
Great, source is outdated now, but I will add this.